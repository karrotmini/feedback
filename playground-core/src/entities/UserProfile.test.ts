import { describe, expect } from 'vitest';

import {
  UserProfile,
  UserProfileID,
} from '../entities';

describe('UserProfile', test => {
  test('should be valid after UserProfileCreated', () => {
    const id = UserProfileID('TEST');
    const userProfile = new UserProfile(id);
    userProfile.$publishEvent({
      aggregateId: id,
      aggregateName: 'UserProfile',
      eventName: 'UserProfileCreated',
      eventDate: new Date('2022-06-10').getTime(),
      eventPayload: {
        name: null,
        profileImageUrl: null,
      },
    });

    expect(userProfile.$valid).toBe(true);
    expect(userProfile.$snapshot).toMatchSnapshot(
      `$snapshot version ${userProfile.snapshotVersion}`,
    );
  });

  test('updateProfile() only affects if there is any change', () => {
    const id = UserProfileID('TEST');
    const name = 'TEST';
    const profileImage = new URL('file:///profileImage');
    const userProfile = UserProfile.create({
      id,
      name,
      profileImage,
    });
    userProfile.$pullEvents();

    userProfile.updateProfile({
      name,
      profileImage,
    });
    expect(userProfile.$pullEvents()).toHaveLength(0);

    userProfile.updateProfile({
      name: 'CHANGED',
      profileImage,
    });
    expect(userProfile.$pullEvents()).toHaveLength(1);
  });
});
