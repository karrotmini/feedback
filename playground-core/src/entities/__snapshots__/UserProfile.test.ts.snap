// Vitest Snapshot v1

exports[`UserProfile > should be valid after UserProfileCreated > $snapshot version 1 1`] = `
{
  "appIds": [],
  "createdAt": 1654819200000,
  "deletedAt": null,
  "name": null,
  "profileImageUrl": null,
}
`;
